### PR TITLE
New harness function for easily getting preview parsed from manifest

### PR DESCRIPTION
**Pull request by joshhacksthings**
Functionality to provide manifest preview in interactive python mode

### CODE REVIEW

**Comment by joshhacksthings** [comment]
hey Strateos team - @VarunKanwar @rhysormond @yangchoo 
just checking if anyone has the bandwidth to review this as this is blocking some of the tooling I'm working on.

File: autoprotocol/harness.py [hunk 1]
```diff
@@ -26,6 +26,23 @@
 }
 
 
+def get_protocol_preview(protocol, name):
```
Comment (yangchoo) [inline autoprotocol/harness.py]: We haven't historically been good about documentation for things in harness, but would you mind adding some documentation here?

Some blurb about what the function does and an example would be awesome :)

File: autoprotocol/harness.py [hunk 2]
```diff
@@ -26,6 +26,23 @@
...
+    manifest_json = io.open('manifest.json', encoding='utf-8').read()
```
Comment (yangchoo) [inline autoprotocol/harness.py]: should probably wrap this in a try/catch in case the file doesn't exist

File: autoprotocol/harness.py [hunk 3]
```diff
@@ -26,6 +26,23 @@
...
+    manifest = Manifest(json.loads(manifest_json))
+
+    try:
+        source = [
+            manifest.protocols[i] for i, m in enumerate(manifest.protocols)
+            if manifest.protocols[i]['name'] == name][0]['preview']
```
Comment (yangchoo) [inline autoprotocol/harness.py]: nit: the formatting is slightly odd here. there's also an implicit picking of the first match, which would be good to call out/document or add an error case for
```
protocols = [
       manifest.protocols[i] for i, m in enumerate(manifest.protocols)
       if manifest.protocols[i]['name'] == name
]
source = protocols[0]['preview'] # or some other check
```

File: autoprotocol/harness.py (hunk 1 above)
Comment (joshhacksthings) [reply autoprotocol/harness.py]: can do!

File: docs/changelog.rst [hunk 4]
```diff
@@ -2,6 +2,9 @@
 Changelog
 =========
 
+* :release:`5.4.2 <2019-06-20>`
```
Comment (yangchoo) [inline docs/changelog.rst]: nit: We usually release as a separate commit. There's no need to include this for now (also, technically this is a not a bugfix, so it should be a minor bump)

File: test/manifest_test.py [hunk 5]
```diff
@@ -605,3 +605,7 @@ def test_cover_type_on_store(self):
         self.protocol.uncover(test2)
         seal_on_store(self.protocol)
         assert (self.protocol.instructions[-1].lid == "low_evaporation")
+
+    def test_get_protocol_preview(self):
+        preview = get_protocol_preview(self.protocol, "TestMethod", manifest="test/manifest_test.json")
+        assert isinstance(preview, dict)
```
Comment (yangchoo) [inline test/manifest_test.py]: Could you match it against the known output instead of just a type check? Thanks!

File: test/manifest_test.py [hunk 6]
```diff
@@ -608,4 +608,8 @@ def test_cover_type_on_store(self):
 
     def test_get_protocol_preview(self):
         preview = get_protocol_preview(self.protocol, "TestMethod", manifest="test/manifest_test.json")
-        assert isinstance(preview, dict)
+        assert "my_string" in preview
```
Comment (yangchoo) [inline test/manifest_test.py]: very minor nit, but could you accumulate these keys into a list and iterate through that instead?

Comment (joshhacksthings) [reply test/manifest_test.py]: Definitely!
So something like:
```
manifest_keys = ["my_string", "my_container", "my_volume", "my_length", "my_bool"]
for key in manifest_keys:
    assert (key in preview)
```

Comment (yangchoo) [reply test/manifest_test.py]: yup, something like that would work

Comment (joshhacksthings) [reply test/manifest_test.py]: great! thanks!

**Pull request merged**